### PR TITLE
Return cursor to start position when immediately exiting visual line mode

### PR DESCRIPTION
**Pull request by caleywoods**
**What this PR does / why we need it**:
This PR allows the cursor to return to its original position when the user enters visual line selection mode and then immediately exits it with `<Esc>`. This replicates VIM's behavior when pressing `V<Esc>`.

**Which issue(s) this PR fixes**
[This issue](https://github.com/VSCodeVim/Vim/issues/3975)

**Special notes for your reviewer**:
I noticed when working on this issue that in VIM when you enter visual line mode with `V` and then you select more lines with `j` for example, the cursor moves to that same column/position on the line below but it looks like VSCodeVIM always jumps to the end of the next line. 

Not sure if that's intended, perhaps it should be filed as another issue.

### CODE REVIEW

**Comment by J-Fields** [comment]
In Vim, you can move your cursor anywhere within the selected region when in visual line mode. VSCode has no concept of a selection that does not have a cursor at the start or end, so we can't faithfully recreate this in a way that wouldn't be confusing, I think. Maybe we should handle more complicated cases than this, but I think `V<Esc>` is a good start, at least.

**Comment by J-Fields** [comment]
Actually, I've rethought this - I think we should put the cursor on the original column, even if they move down or up lines. This is a bit more limited than Vim, but it's more consistent & intuitive than only preserving the column when you immediately exit.

**Comment by caleywoods** [comment]
Cool, will be just a bit more work to fix the tests as I had started doing originally. I can address it in the evenings this week.

File: test/mode/modeVisualLine.test.ts [hunk 1]
```diff
@@ -471,4 +471,26 @@ suite('Mode Visual Line', () => {
       end: ['111', '222_|', ' ', '444_', '555'],
     });
   });
+
+  suite('Cursor behavior when immediately exiting', () => {
+    newTest({
+      title: 'Returns cursor to original position when immediately exiting visual line',
+      start: ['rocinante', 'nauvoo', '|anubis', 'canterbury'],
```
Comment (xconverge) [inline test/mode/modeVisualLine.test.ts]: I like the cut of your jib

Comment (caleywoods) [reply test/mode/modeVisualLine.test.ts]: Remember the cant

**Comment by J-Fields** [comment]
Hey @caleywoods any update on this?

**Comment by caleywoods** [comment]
@J-Fields I'll see if I can sink some time into it this week. IIRC my original change was what we're now after, it had ~20 some odd failing tests. Of those 20, some, if not all, need re-written to respect the new cursor behavior.

Sorry for falling off the map on this one, I've been spending my evenings working through some interview practice books/problems - Thanks for the reminder, have a great week.

**Comment by J-Fields** [comment]
Any update @caleywoods? If you're too busy with other things, I'm happy to take a crack at polishing this up!

**Comment by caleywoods** [comment]
@J-Fields I would appreciate you doing that. The functionality was pretty simple and it seemed like there were roughly ~20 tests that needed updated which had their final cursor position altered by this chance.

**Comment by J-Fields** [comment]
Will do - thanks for getting this off the ground! Hope you have a great new year 🙂

**Pull request closed without merging**